neo/rawio/openephysbinaryrawio.py

```diff
@@ -204,6 +204,32 @@ def _parse_header(self):
                     else:
                         raise ValueError(f'There is no possible labels for this event: {stream_name}')
 
+                    # # If available, use 'states' to compute event duration
+                    if 'states' in d:
+                        states = d["states"]
+                        timestamps = d["timestamps"]
+                        labels = d["labels"]
+                        rising = np.where(states > 0)[0]
+                        falling = np.where(states < 0)[0]
+                        # make sure first event is rising and last is falling
+                        if states[0] < 0:
```

Comment:
What about the case where states == 0 or multiple values in the beginning or end are smaller or bigger than 0 respectively?

Reply:
I thin that `states` are by definition either `+channel_number` or `-channel_number`.

@jsiegle could you confirm that?

Reply:
That's right, `states` can never be zero. However, it is possible that first event can be falling, e.g. if a TTL line was high when the recording started. Same with the last event being rising.

Reply:
That's taken care of :) in that case the first/last event is dropped